fw/http_parser.c

```diff
@@ -3135,6 +3135,36 @@ __req_parse_x_forwarded_for(TfwHttpMsg *hm, unsigned char *data, size_t len)
 }
 STACK_FRAME_NON_STANDARD(__req_parse_x_forwarded_for);
 
+static int
+__req_parse_x_tempesta_cache(TfwHttpMsg *hm, unsigned char *data, size_t len)
+{
+	int r = CSTR_NEQ;
+	__FSM_DECLARE_VARS(hm);
+
+	__FSM_START(parser->_i_st);
+
+	__FSM_STATE(Req_I_X_Tempesta_Cache) {
+		/*
+		 * Tempesta custom header for PURGE:
+		 *	X-Tempesta-Cache = "X-Tempesta-Cache" ":" method
+		 *	method = "get"
+		 */
+		TRY_STR("get", Req_I_X_Tempesta_Cache, I_Tempesta_Cache_get);
+		__FSM_EXIT(CSTR_NEQ);
+	}
+
+	__FSM_STATE(I_Tempesta_Cache_get) {
+		if (IS_CRLF(c)) {
+			// do some processing of PURGE X-Tempesta-Cache request
+			hm->cache_ctl.flags |= TFW_HTTP_CC_CACHE_PURGE;
+			return __data_off(p);
+		}
+		__FSM_EXIT(CSTR_NEQ);
+	}
+done:
+	return r;
+}
```

Comment:
It'd be handy to add appropriate unit test for this header to `t/unit/test_http_parser.c`

Reply:
I work on tests.
common generic tests with requests load work fine. I work on unit tests for splitted through multiple skb requests.

Reply:
I added test to test_http_msg.c but I cannot run tests on my VM. vmalloc and vfree in the tests init halt the kernel. it reports ```BUG: scheduling while atomic: insmod/5080/0x00000200``` and then kernel panic happens.  so I cannot check the test here.  it's not finished yet. I wanted to improve it but couldn't run it somehow. if I figure out a way to get rid of calls that kill the kernel maybe this will work out. I will try.